syntax = "proto3";

package CTGrpc;

service PythonServer {
  rpc GetScenesRPC (File) returns (JsonString) {}
  rpc ToPhraseHintsRPC(PhraseHintRequest) returns (PhraseHintResponse) {}

  rpc GetKalturaChannelEntriesRPC (PlaylistRequest) returns (JsonString) {}
  rpc DownloadKalturaVideoRPC (MediaRequest) returns (File) {}

  rpc GetEchoPlaylistRPC (PlaylistRequest) returns (JsonString) {}
  rpc DownloadEchoVideoRPC (MediaRequest) returns (File) {}

  rpc GetYoutubePlaylistRPC (PlaylistRequest) returns (JsonString) {}
  rpc DownloadYoutubeVideoRPC (MediaRequest) returns (File) {}

  rpc ConvertVideoToWavRPCWithOffset (FileForConversion) returns (File) {}
  rpc ProcessVideoRPC (File) returns (File) {}

  rpc ComputeFileHash (FileHashRequest) returns (FileHashResponse) {}
  rpc GetMediaInfoRPC(File) returns (JsonString) {}
}


// The request message containing the user's name.
message JsonString {
  string json = 1;
}

// The response message containing the greetings.
message PlaylistRequest {
  string Url = 1;

  int32 stream = 2;  
  JsonString metadata = 3;
}

message MediaRequest {
  string videoUrl = 1;
  string additionalInfo = 2;
}
message PhraseHintRequest {
  string rawPhraseData = 1;
}

message File {
  string filePath = 1;
  string ext = 2;
}

message FileForConversion {
  File file = 1;
  float offset = 2;
}

message EPubData {
  string title = 1;
  string author = 2;
  string publisher = 3;
  repeated EPubChapter chapters = 4;
  string file = 5;
}

message EPubChapter {
  string title = 1;
  string text = 2;
  File image = 3;
}

message FileHashRequest {
  string file = 1;
  string algorithms = 2;
}

message FileHashResponse {
  string result = 1;
}

message PhraseHintResponse {
	string result = 1;
}